packages/api/portfolioDrafts/funding/createFundingStep.test.ts

```diff
@@ -0,0 +1,418 @@
+});
+
+describe("Handle service level error", function () {
+  it("should return generic Error if exception caught", async () => {
+    jest.spyOn(console, "error").mockImplementation(() => jest.fn()); // suppress output
+    ddbMock.on(UpdateCommand).rejects("Some error occurred");
+    const result = await handler(validRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(DATABASE_ERROR);
+    expect(result.statusCode).toEqual(ErrorStatusCode.INTERNAL_SERVER_ERROR);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+  });
+});
+
+describe("Path parameter tests", function () {
+  it("should require path param", async () => {
+    const emptyRequest: APIGatewayProxyEvent = {} as any;
+    const result = await handler(emptyRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(PATH_PARAMETER_REQUIRED_BUT_MISSING);
+    expect(result.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+  });
+  it("should return error when path param not UUIDv4 (to avoid attempting update)", async () => {
+    const invalidRequest: APIGatewayProxyEvent = {
+      pathParameters: { portfolioDraftId: "invalid" },
+    } as any;
+    const result = await handler(invalidRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(NO_SUCH_PORTFOLIO_DRAFT);
+    expect(result.statusCode).toEqual(ErrorStatusCode.NOT_FOUND);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+    expect(JSON.parse(result.body).message).toMatch(/Portfolio Draft with the given ID does not exist/);
+  });
+});
+
+describe("Request body tests", function () {
+  it("should return error when request body is empty", async () => {
+    const emptyRequest: APIGatewayProxyEvent = {
+      body: "",
+      pathParameters: { portfolioDraftId: uuidv4() },
+    } as any;
+    const result = await handler(emptyRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(REQUEST_BODY_EMPTY);
+    expect(result.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+    expect(JSON.parse(result.body).message).toMatch(/Request body must not be empty/);
+  });
+  it("should return error when request body is invalid json", async () => {
+    const invalidBodyRequest: APIGatewayProxyEvent = {
+      body: JSON.stringify({ foo: "bar" }) + "}", // invalid json
+      pathParameters: { portfolioDraftId: uuidv4() },
+    } as any;
+    const result = await handler(invalidBodyRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(REQUEST_BODY_INVALID);
+    expect(result.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+    expect(JSON.parse(result.body).message).toMatch(/A valid request body must be provided/);
```

Comment:
Ditto to the comment on redundant clauses here and likely a bunch of places.

Reply:
Resolved in other comments.